packages/clerk-js/src/v4/UserProfile/MfaTOTPPage.tsx

```diff
@@ -0,0 +1,192 @@
+import { TOTPResource } from '@clerk/types';
+import { QRCodeSVG } from 'qrcode.react';
+import React from 'react';
+
+import { useCoreUser } from '../../ui/contexts';
+import { useWizard, Wizard } from '../common';
+import { Button, Col, Spinner, Text } from '../customizables';
+import { ClipboardInput, useCardState, useCodeControl, withCardStateProvider } from '../elements';
+import { CodeForm } from '../elements/CodeForm';
+import { useLoadingStatus } from '../hooks';
+import { handleError, sleep, useFormControl } from '../utils';
+import { FormButtonContainer } from './FormButtons';
+import { NavigateToFlowStartButton } from './NavigateToFlowStartButton';
+import { ContentPage } from './Page';
+import { SuccessPage } from './SuccessPage';
+
+export const MfaTOTPPage = withCardStateProvider(() => {
+  const title = 'Add authenticator app';
+  const wizard = useWizard({});
+
+  return (
+    <Wizard {...wizard.props}>
+      <AddAuthenticatorApp
+        title={title}
+        onContinue={wizard.nextStep}
+      />
+
+      <VerifyTOTP
+        title={title}
+        onVerified={wizard.nextStep}
+      />
+
+      <SuccessPage
+        title={title}
+        text={`Multifactor authentication is now enabled. When signing in, you will need to enter a verification code from this authenticator as an additional step.`}
+      />
+    </Wizard>
+  );
+});
+
+type AddAuthenticatorAppProps = {
+  title: string;
+  onContinue: () => void;
+};
+
+type DisplayFormat = 'qr' | 'uri';
+
+const AddAuthenticatorApp = (props: AddAuthenticatorAppProps) => {
```

Comment:
🙃 
I would put both `AddAuthenticatorApp` and `VerifyTOTP` on different files to reduce the length a bit.